flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/testutils/RestoreTestCompleteness.java

```diff
@@ -0,0 +1,129 @@
+
+                    /** Covered with {@link StreamExecWindowAggregate}. */
+                    add(StreamExecLocalWindowAggregate.class);
+                    add(StreamExecGlobalWindowAggregate.class);
+
+                    /** Covered with {@link StreamExecChangelogNormalize}. */
+                    add(StreamExecDropUpdateBefore.class);
+
+                    /** TODO: Remove after FLINK-33676 is merged. */
+                    add(StreamExecWindowAggregate.class);
+
+                    /** TODO: Remove after FLINK-33805 is merged. */
+                    add(StreamExecOverAggregate.class);
+
+                    /** Ignoring python based exec nodes temporarily. */
+                    add(StreamExecPythonCalc.class);
+                    add(StreamExecPythonCorrelate.class);
+                    add(StreamExecPythonOverAggregate.class);
+                    add(StreamExecPythonGroupAggregate.class);
+                    add(StreamExecPythonGroupTableAggregate.class);
+                    add(StreamExecPythonGroupWindowAggregate.class);
+                }
+            };
+
+    @Test
+    public void testMissingRestoreTest()
+            throws IOException, NoSuchMethodException, InstantiationException,
+                    IllegalAccessException, InvocationTargetException {
+        Map<ExecNodeNameVersion, Class<? extends ExecNode<?>>> versionedExecNodes =
+                ExecNodeMetadataUtil.getVersionedExecNodes();
+
+        Set<ClassPath.ClassInfo> classesInPackage =
+                ClassPath.from(this.getClass().getClassLoader())
+                        .getTopLevelClassesRecursive(
+                                "org.apache.flink.table.planner.plan.nodes.exec.stream")
+                        .stream()
+                        .filter(x -> x.getSimpleName().contains("RestoreTest"))
+                        .collect(Collectors.toSet());
+
+        Set<Class<? extends ExecNode<?>>> execNodesWithRestoreTests = new HashSet<>();
+
+        for (ClassPath.ClassInfo classInfo : classesInPackage) {
+            Class<?> restoreTest = classInfo.load();
+            Method getExecNodeMethod = restoreTest.getMethod("getExecNode");
+            Class<? extends ExecNode<?>> execNode =
+                    (Class<? extends ExecNode<?>>)
+                            getExecNodeMethod.invoke(
+                                    restoreTest.getDeclaredConstructor().newInstance());
+            execNodesWithRestoreTests.add(execNode);
+        }
+
+        for (Map.Entry<ExecNodeNameVersion, Class<? extends ExecNode<?>>> entry :
+                versionedExecNodes.entrySet()) {
+            ExecNodeNameVersion execNodeNameVersion = entry.getKey();
+            Class<? extends ExecNode<?>> execNode = entry.getValue();
+            if (!SKIP_EXEC_NODES.contains(execNode)) {
+                final String msg =
+                        "Missing restore test for "
+                                + execNodeNameVersion
```

Comment:
Is it fine that both the version and the exec node are part of the msg?

Reply:
Yeah I think that is being explicit for developers.